scripts/update-major-version-tag.sh

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+
+set -x
+set -e
+set -o pipefail
+
+RELEASE_VERSION="${1}"
+
+if [[ -z $RELEASE_VERSION ]]; then
+  echo "Error: No release version specified."
+  exit 1
+fi
+
+MAJOR_VERSION_TAG="v${RELEASE_VERSION/\.*/}"
+
+git config user.name github-actions
+git config user.email github-actions@github.com
+
+if git show-ref --tags "$MAJOR_VERSION_TAG" --quiet; then
+  echo "Tag ${MAJOR_VERSION_TAG} exists, attempting to delete it."
+  git tag --delete "$MAJOR_VERSION_TAG"
+  git push --delete origin "$MAJOR_VERSION_TAG"
+else 
+  echo "Tag ${MAJOR_VERSION_TAG} does not exist, creating it from scratch."
+fi
+
+git tag "$MAJOR_VERSION_TAG" HEAD
+git push --tags
```

Comment:
This has the potential to delete the old one and fail to push the new one. We can do this in a single push.
```suggestion
if git show-ref --tags "$MAJOR_VERSION_TAG" --quiet; then
  echo "Tag ${MAJOR_VERSION_TAG} exists, attempting to replace it."
else 
  echo "Tag ${MAJOR_VERSION_TAG} does not exist, creating it from scratch."
fi

git tag -f "$MAJOR_VERSION_TAG" HEAD
git push --tags -f
```

Reply:
wondering if we should keep the check and the feedback? seems less useful to have it + if/else + feedback now that we're doing it all in one go

additionally, maybe we should just push the existing tag instead of all of them? wdyt?

Reply:
> wondering if we should keep the check and the feedback? 
I'm agnostic on this one.

> `git push "$MAJOR_VERSION_TAG" -f`

Even better! 
I'm not 100% on if `--force-with-lease` works as expected (force, but fail if someone else updated since we fetched) but if so that would be even nicer.

Reply:
Suggestion updated with explicit tag and `--force-with-lease`